### ISSUE TITLE
Edit of a PDS or PDSE (as opposed to a member) should fail or show a list of members

### ISSUE THREAD

**Issue by MikeFultonDev** [zopencommunity/vimport#75]
This is opened from a report in: https://github.com/zopencommunity/vimport/issues/68

Edit a PDS by mistake
vim "//'@02858.t.cntl'"

@IgorTodorovskiIBM we probably need more 'handshaking' for this scenario and it is also related to: https://github.com/zopencommunity/vimport/issues/73 but perhaps we at least fail the edit as a starting point.

**Comment by MikeFultonDev** [comment]
@IgorTodorovskiIBM for now, I think a reasonable fix would be for you to check the information on the dataset after the open, and if it's a PDSE or PDS _and_ no member was specified, to fail, e.g.

```
struct DFILE* dsfile = open_dataset(ffname, NULL);
...
if ((dsfile->dsorg == D_PDSE || dsfile->dsorg == D_PDS) && has_member(dsfile->difile)) {
   error
}
...

We need to expose the `has_member` function for this - and probably that can be at the dsfile level. 
I don't think it makes sense to put this into libdio because for other uses, the user may want to read the pds member (e.g. to get the directory information for the pds).

Thoughts?

**Comment by IgorTodorovskiIBM** [comment]
Agree, I have added the code in https://github.com/zopencommunity/vimport/commit/2149fe03b128ac896c6e107743cb48e87c33d1fb and https://github.com/zopencommunity/libdio/commit/fcabd7629983d1de70ab49f3a6d20e681d4e1648. It will be available in the next release of vim%datasetio
